File: project_euler/66-quadratic_diophantine/Solution.java
class Solution {
    public static void main (String[] args) {
        long max_d = 0;
        long max_x = 0;

        for (long d = 1; d < 1000; d++) {
            System.out.println(d);
            if (isLong(Math.sqrt(d))) {continue;}
            long x = 0;
            while (!isLong(y(x, d)) || y(x, d) == 0) {
                x += 1;
            }
            if (x > max_x) {
                max_d = d;
                max_x = x;
            }
        }
        System.out.println(max_x);
        System.out.println(max_d);
        System.out.println(y(max_x, max_d));
        System.out.println(max_x * max_x - max_d * y(max_x, max_d) * y(max_x, max_d));
    }

    public static boolean isLong (double number) {
        return (long) number == number;
    }

    public static double y (long x, long d) {
        return (double) Math.sqrt((double) (x * x - 1) / d);
    }
}

File: project_euler/66-quadratic_diophantine/solution.py
import math

#def y(x, d):
#    return math.sqrt((x ** 2 - 1) / d)

def x(y, d):
    return math.sqrt(1 + d * y ** 2)

def is_int(num):
    return num == int(num)

max_d = 0
max_x = 0
for d in range(1, 1000):
    print(f"d: {d}/1000", end="\r")
    if not math.sqrt(d) - math.isqrt(d) == 0:
        y = 1
        while not is_int(x(y, d)) or x(y, d) == 0:
            y += 1

        if x(y, d) > max_x:
            max_d = d
            max_x = x(y, d)
print(max_d)